jerry-core/ecma/builtin-objects/ecma-builtin-global.cpp

```diff
@@ -878,16 +877,21 @@ ecma_builtin_global_object_encode_uri_helper (ecma_value_t uri, /**< uri argumen
    * and compute the length of the output, then we encode the input.
    */
 
-  lit_utf8_iterator_t iter = lit_utf8_iterator_create (input_start_p, input_size);
-  lit_utf8_size_t output_length = 1;
-  while (!lit_utf8_iterator_is_eos (&iter))
+  lit_utf8_byte_t *input_char_p = input_start_p;
+  lit_utf8_byte_t *input_end_p = input_start_p + input_size;
+  lit_utf8_size_t output_length = 0;
+
+  while (input_char_p < input_end_p)
   {
-    /* Input validation. */
-    lit_code_point_t character = lit_utf8_iterator_read_next (&iter);
+    /*
+     * We expect that the input is a valid UTF-8 sequence,
+     * so we only need to reject stray surrogate pairs.
+     */
 
-    if (character <= 0x7f)
+    /* Input validation. */
+    if (*input_char_p <= LIT_UTF8_1_BYTE_CODE_POINT_MAX)
```

Comment:
Isn't the utf-8 encoding / decoding logic implemented in `lit`? If so, maybe, we can use the `lit`'s helpers in this code.